docs/conceptual/core/codec.md

```diff
@@ -1,13 +1,146 @@
 ---
-title: Parity Scale Codec
+title: SCALE Codec
 ---
 
-* What is parity scale codec?
+The SCALE (Simple Concatenated Aggregate Little-Endian) Codec is a lightweight, efficient, binary serialization and deserialization codec.
 
-* How does it work?
+It is designed for high-performance, copy-free encoding and decoding of data in resource-constrained execution contexts, like the [Substrate runtime](conceptual/runtime/index.md). It is not self-describing in any way and assumes the decoding context has all type knowledge about the encoded data.
 
-* Examples
+## SCALE for Substrate
 
-* Why did we invent our own codec?
+Substrate uses the [`parity-scale-codec`](https://github.com/paritytech/parity-scale-codec), a Rust implementation of the SCALE Codec. This library and the SCALE codec is advantageous for Substrate and blockchain systems because:
 
-* What implementations exist?
\ No newline at end of file
+* It does not use Rust STD, and thus can compile to Wasm.
+* It is built to have great support in Rust for deriving codec logic for new types: just add `#[derive(Encode, Decode)]`.
+
+It is important to define the encoding scheme used on Substrate rather than reuse an existing Rust codec library because this codec needs to be re-implemented on other platforms and languages that want to support interoperability.
+
+## Codec Definition
+
+Here you will find how the SCALE codec encodes different types.
+
+### Fixed-width Integers
+
+Basic integers are encoded using a fixed-width little-endian (LE) format.
+
+#### Example
+
+- `signed 8-bit integer 69`: `0x45`
+- `unsigned 16-bit integer 42`: `0x2a00`
+- `unsigned 32-bit integer 16777215`: `0xffffff00`
+
+### Boolean
+
+Boolean values are encoded using the least significant bit (LSB) of a single byte.
+
+#### Example
+
+- `boolean false`: `0x00`
+- `boolean true`: `0x01`
+
+### Compact/General Integers
```

Comment:
This should go right after (or right before) fixed width integers.

Reply:
moved